### PR TITLE
[dart][dart-dio] Bump Dio to 5.0.0

### LINKED ISSUE #14281: [dart-dio] Support DioX
### Is your feature request related to a problem? Please describe.

Dio has been forked by another group of Flutter developers, with the intention of more active development and maintenance.

https://github.com/flutterchina/dio/issues/1607

An option to generate a DioX Client, or perhaps migrating entirely to DioX would be great.

@kuhnroyal

### PR DESCRIPTION
**Pull request by kuhnroyal**
This PR updates the dart-dio generator to use `dio: ^5.0.0`.

Closes #14281 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] In case you are adding a new generator, run the following additional script : 
  ```
  ./bin/utils/ensure-up-to-date.sh
  ``` 
  Commit all changed files.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (6.3.0) (minor release - breaking changes with fallbacks), `7.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

CC @jaumard (2018/09) @josh-burton (2019/12) @amondnet (2019/12) @sbu-WBT (2020/12) @kuhnroyal (2020/12) @agilob (2020/12) @ahmednfwela (2021/08)

### CODE REVIEW

**Review by ahmednfwela — APPROVED** [review]
LGTM, there is just a small typo

**Review by josh-burton — APPROVED** [review]

**Comment by asaarnak** [comment]
`diox` got ownership of `dio` github and migrated to `dio: 5.0.0`

**Comment by ahmednfwela** [comment]
that's great news, then we can just replace all `Diox` strings in this PR to just `Dio5` and change the `dio_import` back to normal dio

**Comment by kuhnroyal** [comment]
> that's great news, then we can just replace all `Diox` strings in this PR to just `Dio5` and change the `dio_import` back to normal dio

We depend on https://github.com/lomsa-dev/http-mock-adapter/pull/140 since we have tests using this package.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java [hunk 1]
```diff
@@ -109,19 +119,29 @@ public DartDioClientCodegen() {
...
         dateOption.setEnum(dateOptions);
         cliOptions.add(dateOption);
+
+        // Dio implementation Option
+        final CliOption dioOption = CliOption.newString(DIO_LIBRARY, "Specify DIO library");
+        dioOption.setDefault(DIO_LIBRARY_DEFAULT);
+
+        final Map<String, String> dioOptions = new HashMap<>();
+        dioOptions.put(DIO_LIBRARY_DIO, "[DEFAULT] Dio (potentially not maintained anymore)");
+        dioOptions.put(DIO_LIBRARY_DIOX, "DioX (maintained hard-fork of dio with nwer features)");
```
[the remark this answers is not in the document]
Comment (kuhnroyal) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/DartDioClientCodegen.java]: This has been removed

File: docs/generators/dart-dio.md [hunk 2]
```diff
@@ -11,7 +11,7 @@ title: Documentation for the dart-dio Generator
 | generator type | CLIENT | |
 | generator language | Dart | |
 | generator default templating engine | mustache | |
-| helpTxt | Generates a Dart Dio client library with null-safety. | |
+| helpTxt | Generates a Dart Dio(X) client library with null-safety. | |
```
Comment (ahmednfwela) [inline docs/generators/dart-dio.md:14]: remove this change since it's now only one dio

**Review by amondnet — APPROVED** [review]

File: samples/openapi3/client/petstore/dart-dio/oneof_polymorphism_and_inheritance/lib/src/api.dart [hunk 3]
```diff
@@ -27,8 +27,8 @@ class Openapi {
         this.dio = dio ??
             Dio(BaseOptions(
```
Comment (AlexV525) [inline samples/openapi3/client/petstore/dart-dio/oneof_polymorphism_and_inheritance/lib/src/api.dart:28]: Now `dio` 5.0 will not imply the default content type, please ensure it's valid in requests.

File: docs/generators/dart-dio.md (hunk 2 above)
Comment (kuhnroyal) [reply docs/generators/dart-dio.md:14]: Fixed, also removed the null-safety part, this seems obvious now with Dart 3 coming.

File: samples/openapi3/client/petstore/dart-dio/oneof_polymorphism_and_inheritance/lib/src/api.dart (hunk 3 above)
Comment (kuhnroyal) [reply samples/openapi3/client/petstore/dart-dio/oneof_polymorphism_and_inheritance/lib/src/api.dart:28]: We have always manually set the content-type if it is specified in the spec. See https://github.com/OpenAPITools/openapi-generator/blob/master/samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/lib/src/api/fake_api.dart#L147

Comment (kuhnroyal) [reply samples/openapi3/client/petstore/dart-dio/oneof_polymorphism_and_inheritance/lib/src/api.dart:28]: I think thats enough for now.

**Comment by kuhnroyal** [comment]
Tests are not running on min SDK due to https://github.com/lomsa-dev/http-mock-adapter/pull/140/files#r1104810809

**Review by amondnet — APPROVED** [review]

**Pull request merged**